Create operation

AlineLermen opened the pull request.

AlineLermen committed:
WIP initial create operation for 'USO'

AlineLermen committed:
Add create-operation

AlineLermen committed:
Fix error name

lucasA27 commented on src/usecases/create-operation/use-case.ts:
```diff
@@ -0,0 +1,91 @@
+        }
+      }
+      
+      if (payload.operationType == OperationTypeEnum.USO && payload.creditAmount > user.balance) {
```
adicionar === para também comparar o tipo de dado

lucasA27 commented on src/usecases/create-operation/use-case.ts:
```diff
@@ -0,0 +1,91 @@
+      const operation = await this.operationRepository.createOperation(userOperation)
+
+      let newBalance = 0
+      if (payload.operationType == OperationTypeEnum.USO) {
```
o mesmo do de cima

AlineLermen committed:
Adiciona verificacao do tipo de dado no if

AlineLermen committed:
WIP Modifica forma de obter o valor de creditos da operação

Retorna id do pagamento no create-payment e recebe id do pagamento no create-operation de COMPRA

AlineLermen committed:
Busca valor da compra pelo paymentId
